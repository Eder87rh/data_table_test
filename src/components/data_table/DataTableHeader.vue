<template>
    <tr>
        <th v-for="(title, index) in titles" 
            @click="sort(title)"
            :key="index">
            <i v-if="sortBy === title && orientation === 'desc'" 
                class="material-icons md-light">
                arrow_downward
            </i>
            <i v-if="sortBy === title && orientation === 'asc'" 
                class="material-icons md-light">
                arrow_upward
            </i>
            {{ title.toUpperCase() }}
        </th>
    </tr>
</template>

<script>
import { mapState } from 'vuex'
export default {
    props:{
        titles:{
            required:true,
            type:Array,
        }
    },
    data(){
        return{

        }
    },
    methods:{
        sort(title){
            if(this.sortBy === title){
                this.$store.commit('toggleSortOrientation')
            }else{
                this.$store.commit('setSortBy', title)
                this.$store.commit('setSortOrientationDesc')
            }
        }
    },
     computed:{
        ...mapState({
            sortBy: ({modFilters}) => modFilters.sortBy,
            orientation: ({modFilters}) => modFilters.sortOrientation
        })
    }
}
</script>

<style lang="scss" scoped>

</style>
