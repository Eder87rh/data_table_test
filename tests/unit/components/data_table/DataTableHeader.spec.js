import Vuex from 'vuex'
import { shallowMount, createLocalVue } from '@vue/test-utils'
import DataTableHeader from '@/components/data_table/DataTableHeader.vue'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('DataTableHeader.vue',() => {
    it('match snapshot with titles',() => {
        const titles = ['id', 'name', 'description', 'date', 'amount', 'actions']
        const wraper = shallowMount(DataTableHeader, {
            propsData:{
                titles
            },
            mocks: {
                $store: {
                    state: {
                        modFilters: {
                            sortBy: "date",
                            sortOrientation: "desc"
                        }
                    }
                }
            },
            localVue
        })

        expect(wraper.html()).toMatchSnapshot()
    })
})