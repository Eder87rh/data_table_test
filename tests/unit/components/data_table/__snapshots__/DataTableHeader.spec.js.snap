// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableHeader.vue match snapshot with titles 1`] = `
<tr>
  <th>
    <!---->
    <!---->
    ID
  </th>
  <th>
    <!---->
    <!---->
    NAME
  </th>
  <th>
    <!---->
    <!---->
    DESCRIPTION
  </th>
  <th><i class="material-icons md-light">
      arrow_downward
    </i>
    <!---->
    DATE
  </th>
  <th>
    <!---->
    <!---->
    AMOUNT
  </th>
  <th>
    <!---->
    <!---->
    ACTIONS
  </th>
</tr>
`;
